src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -250,12 +245,11 @@ object RuleUtils {
       // can be transformed to 'Project -> Filter -> Logical Relation'. Thus, with transformDown,
       // it will be matched again and transformed recursively which causes stack overflow exception.
       case baseRelation @ LogicalRelation(
-            _ @HadoopFsRelation(location: PartitioningAwareFileIndex, _, _, _, _, _),
+            _ @HadoopFsRelation(baseLocation: FileIndex, _, _, _, _, _),
```

Comment:
could you restrict it to PartitionAwareFileIndex or TahoeFileIndex. There are other FileIndex implementations which we don't support / haven't investigated fully and we don't know what the behavior could be I guess.

Reply:
In that case, an exception will be thrown with pluggable source provider impl.

Reply:
got it. Could you please add the comment in the code then? it's otherwise difficult to guess this part for reviewers